awx/api/serializers.py

```diff
@@ -1038,7 +1038,9 @@ def _update_password(self, obj, new_password):
             # as the modified user then inject a session key derived from
             # the updated user to prevent logout. This is the logic used by
             # the Django admin's own user_change_password view.
-            update_session_auth_hash(self.context['request'], obj)
+            if self.instance and self.context['request'].user.username == obj.username:
```

Comment:
`self.context['request']` is lazily loaded, so this will result in db hit I believe